src/server.rs

```diff
@@ -84,47 +86,23 @@ pub fn listen<S, Req, Resp, E>(new_service: S, addr: SocketAddr, options: Option
     #[cfg(not(feature = "tls"))]
     let acceptor = Acceptor::Tcp;
 
-    match options.reactor {
-        None => {
-            let (tx, rx) = futures::oneshot();
-            REMOTE.spawn(move |handle| {
-                Ok(tx.complete(listen_with(new_service, addr, handle.clone(), acceptor)))
-            });
-            ListenFuture { inner: future::Either::A(rx) }
-        }
-        Some(Reactor::Remote(remote)) => {
-            let (tx, rx) = futures::oneshot();
-            remote.spawn(move |handle| {
-                Ok(tx.complete(listen_with(new_service, addr, handle.clone(), acceptor)))
-            });
-            ListenFuture { inner: future::Either::A(rx) }
-        }
-        Some(Reactor::Handle(handle)) => {
-            ListenFuture {
-                inner: future::Either::B(future::ok(listen_with(new_service,
-                                                                addr,
-                                                                handle,
-                                                                acceptor))),
-            }
-        }
-    }
+    listen_with(new_service, addr, options.handle, acceptor)
 }
 
 /// Spawns a service that binds to the given address using the given handle.
 fn listen_with<S, Req, Resp, E>(new_service: S,
                                 addr: SocketAddr,
                                 handle: Handle,
-                                _acceptor: Acceptor)
-                                -> io::Result<SocketAddr>
+                                _acceptor: Acceptor) -> io::Result<SocketAddr>
     where S: NewService<Request = Result<Req, DeserializeError>,
                         Response = Response<Resp, E>,
-                        Error = io::Error> + Send + 'static,
+                        Error = io::Error> + 'static,
           Req: Deserialize + 'static,
           Resp: Serialize + 'static,
           E: Serialize + 'static
 {
     let listener = listener(&addr, &handle)?;
-    let addr = listener.local_addr()?;
+    let addr = listener.local_addr();
```

Comment:
Good question. I don't know the failure modes of `local_addr` but it might be independent of the rest of server initialization. e.g. maybe a `local_addr` doesn't actually exist for some reason?